From c4f380415c631bce7d71a283bac718a8111c8899 Mon Sep 17 00:00:00 2001
From: Algernon <lixiaodong67@huawei.com>
Date: Wed, 12 Jul 2023 20:01:04 +0800
Subject: [PATCH 212/268] The kabi issue is resolved when the rx_buf_len
 extension ring is used to set parameters or obtain APIs

driver inclusion
category: bugfix
bugzilla: https://gitee.com/src-openeuler/kernel/issues/I8EN49

--------------------------------

Fix it because ethtool_ Get in ops structure_ Ringparam, set_ Kabi problem caused by
changes in ringparam member parameters.

Signed-off-by: Xiaodong Li <lixiaodong67@huawei.com>
---
 include/linux/ethtool.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/linux/ethtool.h b/include/linux/ethtool.h
index 929fba1b3a0d..891d4f8a09ae 100644
--- a/include/linux/ethtool.h
+++ b/include/linux/ethtool.h
@@ -419,6 +419,12 @@ struct ethtool_ops {
 			      struct ethtool_eeprom *, u8 *);
 	int	(*get_coalesce)(struct net_device *, struct ethtool_coalesce *);
 	int	(*set_coalesce)(struct net_device *, struct ethtool_coalesce *);
+#ifdef __GENKSYMS__
+	void    (*get_ringparam)(struct net_device *,
+				 struct ethtool_ringparam *);
+	int     (*set_ringparam)(struct net_device *,
+				 struct ethtool_ringparam *);
+#else
 	void	(*get_ringparam)(struct net_device *,
 				 struct ethtool_ringparam *,
 				 struct kernel_ethtool_ringparam *,
@@ -427,6 +433,7 @@ struct ethtool_ops {
 				 struct ethtool_ringparam *,
 				 struct kernel_ethtool_ringparam *,
 				 struct netlink_ext_ack *);
+#endif
 	void	(*get_pauseparam)(struct net_device *,
 				  struct ethtool_pauseparam*);
 	int	(*set_pauseparam)(struct net_device *,
-- 
2.34.1

